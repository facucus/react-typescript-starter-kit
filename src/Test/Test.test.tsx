import "jest-dom/extend-expect";
import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import Test from "./Test";

afterEach(cleanup);

describe("Testing Test component", () => {
  test("Checking correct Title", () => {
    const { getByTestId } = render(<Test text="hola" />);

    expect(getByTestId("title-text")).toHaveTextContent("Hola Mundo");
  });

  test("Checking correct Title", () => {
    const { container } = render(<Test text="This is my test" />);

    const text = container.querySelector("p");
    expect(text).toHaveTextContent("This is my test");
  });

  test("Match Snapshot", () => {
    const { container } = render(<Test text="hola" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
