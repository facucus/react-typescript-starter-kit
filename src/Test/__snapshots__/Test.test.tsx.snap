// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Test component Match Snapshot 1`] = `
<div>
  <h1
    data-testid="title-text"
  >
    Hola Mundo
  </h1>
  <p>
    hola
  </p>
</div>
`;
